ha/core/controllers/pcs/node_controller.py

```diff
@@ -176,6 +167,77 @@ def status(self, nodeids: list = None) -> dict:
         """
         raise HAUnimplemented("This operation is not implemented.")
 
+    def check_cluster_feasibility(self, nodeid: str) -> dict:
+        """
+            Check whether the cluster is going to be offline after node with nodeid is stopped.
+        Args:
+            nodeid (str): Private fqdn define in conf store.
+        Returns:
+            Dictionary : {"status": "", "msg":""}
+        """
+        self._is_node_in_cluster(node_id=nodeid)
+        offline_nodes, node_list = self._get_offline_nodes()
+        if (len(offline_nodes) + 1) >= ((len(node_list)//2) + 1):
+            return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Stopping the node will cause a loss of the quorum"}
+        else:
+            return {"status": const.STATUSES.SUCCEEDED.value, "output": "", "error": ""}
+
+    def _get_offline_nodes(self):
+        """
+        Get the list offline nodes & all nodes
+        Returns:
+            [tuple]: tuple of offline_nodes & all_nodes
+        """
+        cluster_status_xml = self._execute.run_cmd(const.GET_CLUSTER_STATUS)
+        # create element tree object
+        root = ET.fromstring(cluster_status_xml[0])
+        node_list = []
+        nodes_ids = []
+        # iterate news items
+        for item in root.findall('nodes'):
+            # iterate child elements of item
+            for child in item:
+                if child.attrib['online'] == 'false':
+                    nodes_ids.append(child.attrib['id'])
+                node_list.append(child.attrib['name'])
+        Log.info(f"List of offline node ids in cluster in sorted ascending order: {sorted(nodes_ids)}")
+        return sorted(nodes_ids), node_list
+
+    def create_health_event(self, nodeid: str, event_type: str) -> dict:
```

Comment:
can we move this to system health health event model
we need to create health event at multiple place
if we can reuse it will good

we can have 
class HealthEvent:
       get_template(self, nodeid: str, event_type: str)

Reply:
Created get_health_event_template() in system health.